src/llvm-remove-addrspaces.cpp

```diff
@@ -323,7 +323,7 @@ bool removeAddrspaces(Module &M, AddrspaceRemapFunction ASRemapper)
 
         Function *NF = Function::Create(
                 NFTy, F->getLinkage(), F->getAddressSpace(), Name, &M);
-        // no need to copy attributes here, that's done by CloneFunctionInto
+        NF->copyAttributesFrom(F);
```

Comment:
IIUC, they should not be copied here, as they are in the incorrect module, and will end up getting merged with the correct ones by the CloneFunctionInto pass later (https://github.com/llvm/llvm-project/blame/33017e5a3fa2c3194522565cd0e106a931b072b3/llvm/lib/Transforms/Utils/CloneFunction.cpp#L104)

Reply:
Yeah, I realize the actual problem now: Attributes aren't preserved because the function of interest here is a declaration, not a definition (so we never call `CloneIntoFunction`).